lib/Mojolicious/Controller.pm

```diff
@@ -274,12 +275,12 @@ sub url_for {
 
 sub url_for_asset {
   my ($self, $asset) = @_;
-  return $self->url_for($self->app->static->asset_path($asset));
+  return $asset =~ ABSOLUTE ? $asset : $self->url_for($self->app->static->asset_path($asset));
```

Comment:
This is wrong, in Perl we need to return a `Mojo::URL` object. Just let `url_for` handle it.